spec/src/main/asciidoc/queries.asciidoc

```diff
@@ -116,5 +116,23 @@ type Query {
   ...
 ----
 
-==== Unnamed queries
+==== Void Queries
 
+By it's very nature, query methods must return some value, thus it is considered a deployment error for a method with a
+`void` return type to be annotated with `@Query`. If a void method is annotated with the `@Query` annotation, the
+implementation must prevent the application from starting and should provide a log message indicating that this is not
+allowed.
+
+==== Query Names
+
+The name of a query in the schema is obtained using the following order:
+
+* if the method is annotated with a `@Query` annotation containing a non-empty String for it's value, that String value
+is used as the query name.
+* if the method is annotated with a `@Name` annotation containing a non-empty String for it's value, that String value
+is used as the query name.
+* if the method is annotated with a `@JsonbProperty` annotation containing a non-empty String for it's value, that String
+value is used as the query name.
+* if no other name can be determined, the Java method name is used as the query name.
+
+Note that it is considered a deployment error if more than one query method has the same name with the same arguments.
```

Comment:
Same as above. A specific exception or just a general Runtime Exception ?

Reply:
Does this not also apply to `@Mutation` ?

Reply:
I think it is very similar for `@Mutation`.  `@Query` is essentially a field - and fields can be named by `@JsonbProperty`.  I don't think `@Mutations` are fields (maybe input fields?).  Definitely worth investigating.  Good question!